projectq/ops/_basics.py

```diff
@@ -295,23 +295,29 @@ def get_merged(self, other):
 
     def __eq__(self, other):
         """ Return True if same class and same rotation angle. """
-        tolerance = 1.e-12
-        if isinstance(other, self.__class__):
-            difference = abs(self._angle - other._angle)
-            if difference < tolerance:
-                return True
-            else:  # if one angle is close to 0 and the other close to 4*pi:
-                if difference > 4 * math.pi - tolerance:
-                    return True
-                else:
-                    return False
-        else:
+        if not isinstance(other, self.__class__):
             return False
+        difference = abs(_signed_mod(self._angle - other._angle, 4 * math.pi))
```

Comment:
What is the advantage of this construction?
I don't think it simplifies this function. Calculating a signed mod but then taking the abs()...